client/src/components/Input/TextChat.jsx

```diff
@@ -29,6 +29,70 @@ export default function TextChat({ isSearchView = false }) {
   const isNotAppendable = latestMessage?.unfinished & !isSubmitting || latestMessage?.error;
   const { conversationId, jailbreak } = conversation || {};
 
+  const [isSpeechSupported, setIsSpeechSupported] = useState(false);
```

Comment:
This component already has too much complexity I'm trying to refactor out. Use a react hook instead: https://chat.openai.com/share/112d8ec7-bb62-460b-adac-2a45fa60c712

Reply:
I moved the speech recognition to another file.